src/OrleansRuntime/Streams/PersistentStream/PersistentStreamPullingAgent.cs

```diff
@@ -509,6 +509,10 @@ private void StartInactiveCursors(StreamConsumerCollection streamData)
                     }
                     catch (Exception exc)
                     {
+                        if (consumerData.Cursor != null)
+                        {
+                            consumerData.Cursor.RecordDeliveryFailure();
```

Comment:
maybe move that line into `ErrorProtocol` below?
Or at least put it next to `ErrorProtocol` call, so all error handling code is close by?